awswrangler/opensearch/_utils.py

```diff
@@ -28,22 +38,122 @@ def _get_version_major(client: OpenSearch) -> Any:
 
 
+def _is_serverless(client: OpenSearch) -> bool:
+    return getattr(client, "_serverless", False)
+
+
+def _get_service(endpoint: str) -> str:
+    return "aoss" if "aoss.amazonaws.com" in endpoint else "es"
+
+
 def _strip_endpoint(endpoint: str) -> str:
     uri_schema = re.compile(r"https?://")
     return uri_schema.sub("", endpoint).strip().strip("/")
 
 
-def _is_https(port: int) -> bool:
+def _is_https(port: Optional[int]) -> bool:
     return port == 443
 
 
+def _get_default_encryption_policy(collection_name: str, kms_key_arn: Optional[str]) -> Dict[str, Any]:
+    policy: Dict[str, Any] = {
+        "Rules": [
+            {
+                "ResourceType": "collection",
+                "Resource": [
+                    f"collection/{collection_name}",
+                ],
+            }
+        ],
+    }
+    if kms_key_arn:
+        policy["KmsARN"] = kms_key_arn
+    else:
+        policy["AWSOwnedKey"] = True
+    return policy
+
+
+def _get_default_network_policy(collection_name: str, vpc_endpoints: Optional[List[str]]) -> List[Dict[str, Any]]:
+    policy: List[Dict[str, Any]] = [
+        {
+            "Rules": [
+                {
+                    "ResourceType": "dashboard",
+                    "Resource": [
+                        f"collection/{collection_name}",
+                    ],
+                },
+                {
+                    "ResourceType": "collection",
+                    "Resource": [
+                        f"collection/{collection_name}",
+                    ],
+                },
+            ],
+            "Description": f"Default network policy for collection '{collection_name}'.",
+        }
+    ]
+    if vpc_endpoints:
+        policy[0]["SourceVPCEs"] = vpc_endpoints
+    else:
+        policy[0]["AllowFromPublic"] = True
+    return policy
+
+
+def _create_security_policy(
+    collection_name: str,
+    policy: Optional[Union[Dict[str, Any], List[Dict[str, Any]]]],
+    policy_type: str,
+    client: boto3.client,
+    **kwargs: Any,
+) -> None:
+    if not kwargs:
+        kwargs = {}
+    if policy_type == "encryption" and not policy:
+        policy = _get_default_encryption_policy(collection_name, kwargs.get("kms_key_arn"))
+    elif policy_type == "network" and not policy:
+        policy = _get_default_network_policy(collection_name, kwargs.get("vpc_endpoints"))
+    else:
+        raise exceptions.InvalidArgument(f"Invalid policy type '{policy_type}'.")
+    try:
+        client.create_security_policy(
+            name=f"{collection_name}-{policy_type}-policy",
+            policy=json.dumps(policy),
+            type=policy_type,
+            description=f"Default {policy_type} policy for collection '{collection_name}'.",
+        )
+    except botocore.exceptions.ClientError as error:
+        if error.response["Error"]["Code"] == "ConflictException":
+            raise exceptions.Conflict("The policy name or rules conflict with an existing policy.") from error
```

Comment:
Maybe change this to `ResourceNamingConflict` or something like that?

Reply:
Done